nevergrad/functions/images/core.py

```diff
@@ -29,34 +33,131 @@ def __init__(self, problem_name: str = "recovering", index: int = 0) -> None:
         image = PIL.Image.open(path).resize((self.domain_shape[0], self.domain_shape[1]), PIL.Image.ANTIALIAS)
         self.data = np.asarray(image)[:, :, :3]  # 4th Channel is pointless here, only 255.
-
-        # if problem_name == "adversarial":
-        #     assert index <= 100  # If we have that many target images.
-        #     self.data = ..... (here we load the imagee correspnding to index and problem_name; this is
-        #         # the attacked image.)
-
-        array = ng.p.Array(init=128 * np.ones(self.domain_shape), mutable_sigma=True,)
+        # parametrization
+        array = ng.p.Array(init=128 * np.ones(self.domain_shape), mutable_sigma=True)
         array.set_mutation(sigma=35)
         array.set_bounds(lower=0, upper=255.99, method="clipping", full_range_sampling=True)
         max_size = ng.p.Scalar(lower=1, upper=200).set_integer_casting()
         array.set_recombination(ng.p.mutation.Crossover(axis=(0, 1), max_size=max_size)).set_name("")  # type: ignore
+
         super().__init__(self._loss, array)
         self.register_initialization(problem_name=problem_name, index=index)
         self._descriptors.update(problem_name=problem_name, index=index)
 
     def _loss(self, x: np.ndarray) -> float:
+        assert self.problem_name == "recovering"
         x = np.array(x, copy=False).ravel()
         x = x.reshape(self.domain_shape)
         assert x.shape == self.domain_shape, f"Shape = {x.shape} vs {self.domain_shape}"
-
         # Define the loss, in case of recovering: the goal is to find the target image.
-        assert self.problem_name == "recovering"
         assert self.index == 0
-        value = np.sum(np.fabs(np.subtract(x, self.data)))
-
-        # Here we should implement "adversarial" and others.
+        value = float(np.sum(np.fabs(np.subtract(x, self.data))))
         return value
+
+
+# TO BE MOVED TO A SEPARATE FILE
+
+class Normalize(nn.Module):
+    def __init__(self, mean, std):
+        super().__init__()
+        self.mean = torch.Tensor(mean)
+        self.std = torch.Tensor(std)
+
+    def forward(self, x):
+        return (x - self.mean.type_as(x)[None, :, None, None]) / self.std.type_as(x)[None, :, None, None]
+
+
+class Resnet50(nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.norm = Normalize(mean=[0.485, 0.456, 0.406],
+                              std=[0.229, 0.224, 0.225])
+        self.model = resnet50(pretrained=True)
+
+    def forward(self, x):
+        return self.model(self.norm(x))
+
+
+class TestClassifier(nn.Module):
+    def __init__(self, image_size: int = 224):
+        super().__init__()
+        self.model = nn.Linear(image_size * image_size * 3, 10)
+
+    def forward(self, x):
+        return self.model(x.flatten(x.shape[0], -1))
+
+
+# pylint: disable=too-many-arguments
+class ImageAdversarial(base.ExperimentFunction):
+
+    def __init__(
+        self,
+        classifier: nn.Module,
```

Comment:
Most arguments should have a default value, so that the benchmark can run create "ImageAdversarial()" and get an objective function; or maybe just one integer, which is the index of the attacked image in ImageNet.

Reply:
No I would say no need:
- the aim for Laurent is to be able to plug anything inside it, so let's leave it this way and avoid weird behavior with default values which do not match some other inputs etc
- it would not work on benchmarks anyway since we cant get a descriptor for a module etc
- there's a `from_testbed` classmethod where we can define testbeds from a name. This way they can be used for test beds easily.

Reply:
Ok for no default value.
But we should really include this in benchmarks.py, and the user should not have to create the classifier and the images. "from_testbed" is ok (I had in mind something similar, like a string that would be the name of the testcase and would build all parameters -- from_testbed is quite analogous to that),

Reply:
it's exactly that actually ;) with a few more (default) parameters